app/src/main/java/com/iti/android/team1/ebuy/ui/productsScreen/viewmodel/ProductsViewModel.kt

```diff
@@ -50,4 +59,17 @@ class ProductsViewModel(private val repoInterface: IRepository) : ViewModel() {
             }
         }
     }
+
+    fun addProductToFavorite(product: FavoriteProduct) {
+        viewModelScope.launch(Dispatchers.IO) {
+            _resultOfAddingProductToFavorite.postValue(repoInterface.addProductToFavorite(product))
```

Comment:
check here and emit Result State model 
because should not know about DataBaseResponse, which makes tightly coupled